### DISCUSSION TITLE
How to create a hook that sends an email to the new registered user

### DISCUSSION THREAD

**Discussion by ssthiago** [directus/directus#7731] [Archive]
First I would like to thank you for the beautiful work done by you, Directus is really phenomenal.

Well, I'm studying flutter and developing an app as part of my studies. Luckily I bumped into Directus on my way and I'm having success using it as an API to feed my app.

That said, I'm facing the following problem: When registering a new user through my app, I do it through a public role with the following field preset:
```
{
    "role": "my public role ID",
    "status": "draft"
}
```

My idea is, after registering a new user from my app, a message like: 'Check your email to finalize your registration'.

Now, I need to create a hook that sends an email to the new registered user. This email would contain a link capable of changing the status from 'draft' to 'active'. Unfortunately for me, I have no experience with node.js and the hook documentation was not enough for my modest understanding. Could someone give me a light on which way to go?

Thanks again.

**Comment by rijkvanzanten** [comment]
Hi @ssthiago! 

The "full example" on the Hooks guide page actually does send an email too! (https://docs.directus.io/guides/api-hooks/#full-example) Did you see that example?

> Could someone give me a light on which way to go?

Another way would be to simply rely on the existing system invite endpoint, that already does exactly what you're looking for as far as I understand 🙂 https://docs.directus.io/reference/api/system/users/#invite-a-new-user

**Reply by ssthiago** [reply]
Thanks for the answer. I found a thread that is exactly what I need.
[https://github.com/directus/directus/discussions/5583](url)

**Reply by rijkvanzanten** [reply]
Perfect!
